feat(kafka,feed): публикация/обработка событий постов и обновление фида в Redis

SAM19991 opened the pull request:
- добавлен AppKafkaProperties и конфигурация Kafka (Producer/Consumer, ListenerFactory, автосоздание топика через NewTopic)
- добавлен DTO PostCreatedKafkaEvent
- реализован KafkaPostProducer: отправка события после успешного сохранения поста
- реализован KafkaPostConsumer с @KafkaListener и ручным ack: обновление фидов подписчиков в Redis с учетом лимита из конфигов
- интеграция в PostServiceImpl: публикация события в Kafka после publish
- настройка Spring Kafka через application.yaml (bootstrap-servers, сериализация/десериализация, group-id), без хардкода в коде
- build.gradle.kts: добавлена зависимость spring-kafka
- FollowServiceClient: метод для получения списка подписчиков автора

SAM19991 committed:
feat(kafka,feed): публикация/обработка событий постов и обновление фида в Redis
- добавлен AppKafkaProperties и конфигурация Kafka (Producer/Consumer, ListenerFactory, автосоздание топика через NewTopic)
- добавлен DTO PostCreatedKafkaEvent
- реализован KafkaPostProducer: отправка события после успешного сохранения поста
- реализован KafkaPostConsumer с @KafkaListener и ручным ack: обновление фидов подписчиков в Redis с учетом лимита из конфигов
- интеграция в PostServiceImpl: публикация события в Kafka после publish
- настройка Spring Kafka через application.yaml (bootstrap-servers, сериализация/десериализация, group-id), без хардкода в коде
- build.gradle.kts: добавлена зависимость spring-kafka
- FollowServiceClient: метод для получения списка подписчиков автора

mysosnovskaya commented on src/main/java/faang/school/postservice/publisher/kafka/KafkaPostProducer.java:
```diff
@@ -0,0 +1,52 @@
+        List<Long> followerIds = safeList(followServiceClient.getFollowerIds(post.getAuthorId()));
+        long publishedAt = post.getPublishedAt() == null
+                ? System.currentTimeMillis()
+                : post.getPublishedAt().toInstant(ZoneOffset.UTC).toEpochMilli();
```
Это не логика продюсера, он должен только отправлять ивент, а вот что отправлять - ему должно прийти в параметры. Т.е. здесь метод должен принимать post и followerIds. Ну и у созданного поста publishedAt обязательно будет пустым, а у опубликованного поста обязательно будет заполнен. Мы шлем ивент только тогда, когда пост опубликован. Нужно с этой терминологией разобраться и убрать эту проверку)

mysosnovskaya reviewed (approved):
На сертификат зачет

SAM19991 committed:
- Вынесена бизнес-логика из KafkaPostProducer: продюсер больше не дергает FollowServiceClient и не решает, что отправлять — только публикует событие.
- Метод KafkaPostProducer.publishPostCreated теперь принимает (Post post, ListfollowerIds).
- Убран fallback по времени публикации: больше не используем System.currentTimeMillis(); берём publishedAt из поста (ивент шлём только для опубликованного поста).
- PostServiceImpl: внедрён FollowServiceClient; в publishPost получаем followerIds и передаём их в продюсер.
- Обновлены места вызова продюсера.
